src/Transport/DelayDelivery/DelayedMessagesPoller.cs

```diff
@@ -150,15 +150,20 @@ async Task SpinOnce(CancellationToken cancellationToken)
                 return;
             }
 
-            var stopwatch = Stopwatch.StartNew();
-
+            Stopwatch stopwatch = null;
             foreach (var delayedMessage in delayedMessages)
             {
                 if (cancellationToken.IsCancellationRequested)
                 {
                     return;
                 }
 
+                // only allocate if needed
+                if (stopwatch == null)
+                {
+                    stopwatch = Stopwatch.StartNew();
+                }
```

Comment:
I'd suggest `stopwatch = stopwatch ?? Stopwatch.StartNew();` but don't think we can do it with c# language constrained to version 7.3.

How about this:

```suggestion
                stopwatch = stopwatch ?? Stopwatch.StartNew();
```